rust/main/chains/hyperlane-sealevel/src/interchain_gas.rs

```diff
@@ -251,7 +251,17 @@ impl Indexer<InterchainGasPayment> for SealevelInterchainGasPaymasterIndexer {
                 if igp_account_filter == sealevel_payment.igp_account_pubkey {
                     payments.push((sealevel_payment.payment, sealevel_payment.log_meta));
                 } else {
-                    tracing::debug!(sealevel_payment=?sealevel_payment, igp_account_filter=?igp_account_filter, "Found interchain gas payment for a different IGP account, skipping");
+                    tracing::debug!(sealevel_payment=?sealevel_payment, igp_account_filter=?igp_account_filter, "Found interchain gas payment for a different IGP account, neutralizing payment");
+
+                    let non_matching_payment = InterchainGasPayment {
```

Comment:
getting a little hard to read, could we change the logic around here starting with line 249 to be something like:
```
if let Ok(mut sealevel_payment) = self.get_payment_with_sequence(nonce.into()).await {
  // Comment explaining why we do this
  if sealevel_payment.igp_account_pubkey != self.igp.igp_account {
    sealevel_payment.gas_amount = U256::zero();
    sealevel_payment.payment = U256::zero();
  }
  payments.push((sealevel_payment.payment, sealevel_payment.log_meta));
}
```